### DISCUSSION TITLE
OTLP Instrumetation

### DISCUSSION THREAD

**Discussion by cicciodifranco** [celery/kombu#1810] [Q&A]
Hello folks!!! 
Is there a way to instrument `kombu`  using OTLP?

Thank you!

**Comment by auvipy** [comment]
can you please share more detail and context, please?

**Reply by cicciodifranco** [reply]
Sure!! We want to integrate OpenTelemetry distributed tracing. In order to make it work in a distributed environment the queue client should be instrumented to send the traceId along with the message.
For example [here](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation/opentelemetry-instrumentation-pika/src/opentelemetry/instrumentation/pika) you can find the instrumentation for Pika.

Our end goal is to trace a request that starts from a flask application and goes through our microservices.

**Reply by auvipy** [reply]
I saw the code. is there any documentation for that?
